cmd/node/main.go

```diff
@@ -766,11 +774,12 @@ func startStatisticsMonitor(file *os.File, config config.ResourceStatsConfig, lo
 	return nil
 }
 
-func createApiResolver() (facade.ApiResolver, error) {
+func createApiResolver(vmAccountsDB vmcommon.BlockchainHook) (facade.ApiResolver, error) {
 	//TODO replace this with a vm factory
-	vm := &mock.VMExecutionHandlerStub{}
+	cryptoHook := hooks.NewVMCryptoHook()
+	ieleVM := endpoint.NewElrondIeleVM(vmAccountsDB, cryptoHook, ielecommon.Default)
```

Comment:
Why do we create a new VM here and another one for `NewSmartContractProcessor` in `structs.go`?

Reply:
Because we need 2 instances: one for the tx processor and the other one for resolving routes requests (get methods). Since there is a state when executing the vm code, we could not use the same instance as it would corrupt the state when 2 calls would have been made in the same time.